### PR TITLE
[FCE-524] Investigate screen broadcast on iOS after updating EXPO

### PR DESCRIPTION
**Pull request by MiloszFilimowski**
## Description

This is mostly the old plugin, which was updated to work with our new setup. 

## Motivation and Context

Creating iOS native files was broken after upgrading to a new expo version. 

## How has this been tested?

- Successfully run clean prebuild and launch iOS app. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/fishjam-chat/app.json [hunk 1]
```diff
@@ -49,9 +49,13 @@
...
           "android": {
             "enableForegroundService": true
+          },
+          "ios": {
+            "enableScreensharing": true
           }
         }
-      ]
+      ],
+      ["./plugin/build/development-pods-plugin.js"] // Only relevant for local development of packages/ios-client
```
[the remark this answers is not in the document]
Comment (MiloszFilimowski) [reply examples/fishjam-chat/app.json:58]: AFAIK expo uses [JSON5](https://json5.org/) under the hood for app.json so it should be perfectly fine.

File: README.md [hunk 2]
```diff
@@ -23,12 +23,14 @@ room client. To run the app:
...
 3. ```
-   yarn
-   yarn build
-   cd `examples/fishjam-chat`
-   yarn prebuild
-   yarn run android/ios
+      yarn
+      yarn build
+      cd `examples/fishjam-chat`
+      npx expo prebuild --clean
```
Comment (mironiasty) [inline README.md:29]: why this change? 🤔

File: examples/fishjam-chat/package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "android": "expo run:android",
     "ios": "expo run:ios",
     "test": "jest",
-    "build": "tsc",
+    "build": "tsc && tsc -p ./plugin",
```
Comment (mironiasty) [inline examples/fishjam-chat/package.json:11]: What does `-p` options does? I can't find it in [official docs](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

File: examples/fishjam-chat/app.json (hunk 1 above)
Comment (mironiasty) [inline examples/fishjam-chat/app.json:58]: WDHY about having this plugin in package?
(might not be part of release in future)
This way, if we would create another example app, we can easily reuse it

File: packages/react-native-client/plugin/src/withFishjamIOS.ts [hunk 4]
```diff
@@ -0,0 +1,261 @@
+// ios-related code was mostly copied from OneSignal expo plugin: https://github.com/OneSignal/onesignal-expo-plugin/blob/main/onesignal/withOneSignalIos.ts
```
Comment (mironiasty) [inline packages/react-native-client/plugin/src/withFishjamIOS.ts:1]: WDYT about adding a little info in comment somewhere about what this plugin does?

File: README.md (hunk 2 above)
Comment (MiloszFilimowski) [reply README.md:29]: What change do you mean? The `npx expo prebuild` or the removed `yarn build` step?

File: examples/fishjam-chat/package.json (hunk 3 above)
Comment (MiloszFilimowski) [reply examples/fishjam-chat/package.json:11]: It's the same as `--project`. I'm surprised it's not mentioned in the docs however it's there when you run `tsc --help`.

File: examples/fishjam-chat/app.json (hunk 1 above)
Comment (MiloszFilimowski) [reply examples/fishjam-chat/app.json:58]: What other project do you mean? Maybe when we have another one let's think about it then? For playground, not releasable projects it's easier to just copy.

File: packages/react-native-client/plugin/src/withFishjamIOS.ts (hunk 4 above)
Comment (MiloszFilimowski) [reply packages/react-native-client/plugin/src/withFishjamIOS.ts:1]: I added some additional comments on what each step does.

File: README.md (hunk 2 above)
Comment (mironiasty) [reply README.md:29]: changing `yarn prebuild` to `npx expo prebuild`

File: examples/fishjam-chat/package.json (hunk 3 above)
Comment (mironiasty) [reply examples/fishjam-chat/package.json:11]: Hmm, maybe lets use `--project` then? 🤔 
(or leave as is, no strong opinion)

Comment (MiloszFilimowski) [reply examples/fishjam-chat/package.json:11]: Sure!

File: README.md (hunk 2 above)
Comment (MiloszFilimowski) [reply README.md:29]: It's discussed as a "NOTE" below:
> Be sure to run `npx expo prebuild --clean` and not `yarn prebuild --clean` as there's an issue with path generation for ios/.xcode.env.local
